ppx/ast_derive/ast_derive_abstract.ml

```diff
@@ -151,13 +151,172 @@ let handleTdcl light (tdcl : Parsetree.type_declaration) :
 let handleTdclsInStr ~light _rf tdcls =
   List.fold_right
     ~f:(fun tdcl sts ->
-      match handleTdcl light tdcl with
-      | value_descriptions -> List.map ~f:Str.primitive value_descriptions @ sts)
+      let value_descriptions = handleTdcl ~light tdcl in
+      List.map ~f:Str.primitive value_descriptions @ sts)
     tdcls ~init:[]
 
 let handleTdclsInSig ~light _rf tdcls =
   List.fold_right
     ~f:(fun tdcl sts ->
-      match handleTdcl light tdcl with
-      | value_descriptions -> List.map ~f:Sig.value value_descriptions @ sts)
+      let value_descriptions = handleTdcl ~light tdcl in
+      List.map ~f:Sig.value value_descriptions @ sts)
+    tdcls ~init:[]
+
+let handleTdclCstr (tdcl : Parsetree.type_declaration) :
```

Comment:
This function seems to be mostly copied from `handleTdcl` above. Is there a way to reuse more? Or because the plan is to remove the other one, doesn't matter?